core/src/main/java/org/apache/iceberg/avro/SchemaToType.java

```diff
@@ -38,6 +43,11 @@ class SchemaToType extends AvroSchemaVisitor<Type> {
     }
   }
 
+  SchemaToType(Schema root, boolean deriveNameMapping) {
+    this(root);
+    this.deriveNameMapping = deriveNameMapping;
```

Comment:
I don't think this PR should build a name mapping. That can be added in a later PR, and it should not use a custom Avro property.

Where possible, we avoid mixing jobs in the Iceberg project. This class converts a schema from Avro to Iceberg and should do only that. If you want to derive a mapping, I'd recommend building a visitor to do that.